### PR TITLE
connector/oidc: fix hosted domain support.

### LINKED ISSUE #999: Bugs in Google OIDC HD (Hosted Domain) claim
Hi guys,

Please see my comments for this already merged PR: https://github.com/coreos/dex/pull/974

I'm afraid there's a bug in the new 'Google hosted domain support' feature.

### PR DESCRIPTION
**Pull request by rithujohn191**
fixes #999

### CODE REVIEW

**Comment by alonl** [comment]
May also fix the example:
https://github.com/coreos/dex/pull/974#discussion_r128779469

:)

**Comment by ericchiang** [comment]
So quick thought on this.

The hosted domain concept is Google specific, not related to OpenID Connect. There are a ton of features that have been requested that apply to Google and aren't applicable to general oidc providers (e.g. groups support). Why arent we just split this into a unique "google" connector instead of pushing Google features into the oidc connector?

**Comment by rithujohn191** [comment]
Created a new issue for this (https://github.com/coreos/dex/issues/1001). Closing out this PR

**Comment by alonl** [comment]
But... This feature is still broken... Better remove it then, I believe. :)
@ericchiang
@rithujohn191

**Comment by rithujohn191** [comment]
New plan....going to fix this feature for now and then in the next release pull it out into its own Google connector :)

File: examples/config-dev.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ connectors:
 #     clientID: $GOOGLE_CLIENT_ID
 #     clientSecret: $GOOGLE_CLIENT_SECRET
 #     redirectURI: http://127.0.0.1:5556/dex/callback
-#     hostedDomain: $GOOGLE_HOSTED_DOMAIN
+#     hostedDomains: $GOOGLE_HOSTED_DOMAIN
```
Comment (alonl) [inline examples/config-dev.yaml]: It's an array. I think it should be:
```
hostedDomains:
- $GOOGLE_HOSTED_DOMAIN
```

**Comment by rithujohn191** [comment]
@ericchiang if this looks ok could you approve the PR? Thanks

**Comment by rubbish** [comment]
@rithujohn191 I was just wondering how soon a release could be cut with this in it? I have a need to restrict logins to a certain gsuite domain. Thanks!

**Comment by alonl** [comment]
@rubbish this won't help you. Notice this sentence from the [Google oidc documentation](https://developers.google.com/identity/protocols/OpenIDConnect):

> Don't rely on this UI optimization to control who can access your app, as client-side requests can be modified. Be sure to validate that the returned ID token has an hd claim value that matches what you expect (e.g. mycolledge.edu). Unlike the request parameter, the ID token claim is contained within a security token from Google, so the value can be trusted.

**Comment by alonl** [comment]
It may be worthwhile to emphasize this in dex as well.

**Comment by rubbish** [comment]
@alonl This PR does check the hd claim in the id token: https://github.com/coreos/dex/pull/1000/files#diff-dbaa70d509cfdb00b729984f27415946R201 So it would validate for those configured hosted domains.

**Comment by alonl** [comment]
Oh, you're right. That's just the issue I reported myself. Sorry for the confusion.

**Pull request merged**